components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/processors/SPInitLogoutRequestProcessor.java

```diff
@@ -134,12 +134,24 @@ public SAMLSSOReqValidationResponseDTO process(LogoutRequest logoutRequest, Stri
                                     "Issuer name :" + splitIssuer[0] + " " + "has being spilt");
                         }
                     } else {
+                        SAMLSSOServiceProviderDO serviceProvider = sessionInfoData.getServiceProviderList().get(issuer);
+                        if (serviceProvider != null) {
+                            SAMLSSOUtil.setTenantDomainInThreadLocal(
+                                    sessionInfoData.getServiceProviderList().get(issuer).getTenantDomain());
```

Comment:
Why not use the 'serviceProvider' reference retrieved above rather than making the method call again